kratos/processes/apply_periodic_boundary_condition_process.cpp

```diff
@@ -0,0 +1,362 @@
+    }
+
+    /**
+     * @brief Function initializes the solution step
+     */
+    void ApplyPeriodicConditionProcess::ExecuteInitializeSolutionStep()
+    {
+    }
+
+    /**
+     * @brief Function to print the information about this current process
+     */
+    void ApplyPeriodicConditionProcess::PrintInfo(std::ostream& rOStream) const
+    {
+        rOStream <<"ApplyPeriodicConditionProcess Process "<<std::endl;
+    }
+
+    template <int TDim>
+    void ApplyPeriodicConditionProcess::ApplyConstraintsForPeriodicConditions()
+    {
+        const double start_apply = OpenMPUtils::GetCurrentTime();
+        const int num_vars = mParameters["variable_names"].size();
+        BinBasedFastPointLocatorConditions<TDim> bin_based_point_locator(mrMasterModelPart);
+        bin_based_point_locator.UpdateSearchDatabase();
+
+        const int num_slave_nodes = mrSlaveModelPart.NumberOfNodes();
+        const NodeIteratorType it_slave_node_begin = mrSlaveModelPart.NodesBegin();
+
+        IndexType num_slaves_found = 0;
+
+        #pragma omp parallel for schedule(guided, 512) reduction( + : num_slaves_found )
+        for(int i_node = 0; i_node<num_slave_nodes; ++i_node)
+        {
+            Condition::Pointer p_host_cond;
+            VectorType shape_function_values;
+            NodeIteratorType it_slave_node = it_slave_node_begin;
+            std::advance(it_slave_node, i_node);
+            array_1d<double, 3 > transformed_slave_coordinates;
+            TransformNode(it_slave_node->Coordinates(), transformed_slave_coordinates);
+
+            // Finding the host element for this node
+            const bool is_found = bin_based_point_locator.FindPointOnMeshSimplified(transformed_slave_coordinates, shape_function_values, p_host_cond, mSearchMaxResults, mSearchTolerance);
+            if(is_found)
+            {
+                ++num_slaves_found;
+                for (int j = 0; j < num_vars; j++)
+                {
+                    const std::string var_name = mParameters["variable_names"][j].GetString();
+                    // Checking if the variable is a vector variable
+                    if (KratosComponents<VariableComponent<VectorComponentAdaptor<array_1d<double, 3>>>>::Has(var_name + "_X"))
+                    {   // TODO: Look for a better alternative to do this.
+                        ConstraintSlaveNodeWithConditionForVectorVariable<TDim>(*it_slave_node, p_host_cond->GetGeometry() , shape_function_values, var_name);
+                    } else if (KratosComponents<VariableType>::Has(var_name))
+                    {
+                        ConstraintSlaveNodeWithConditionForScalarVariable<TDim>(*it_slave_node, p_host_cond->GetGeometry() , shape_function_values, var_name);
+                    }
+                }
+            }
+        }
+        KRATOS_WARNING_IF("",num_slaves_found != mrSlaveModelPart.NumberOfNodes())<<"Periodic condition cannot be applied for all the nodes."<<std::endl;
```

Comment:
the label should not be empty => "ApplyPeriodicConditionsProcess" or sth like this

Reply:
done !